### PR TITLE
*: update dependency and tools

### PR DESCRIPTION
**Pull request by BusyJay**
- update bindgen to 0.69 to keep compatible with latest llvm
- update rand to 0.8
- update clap to 4
- update prost to 0.12
- fix clippy warnings (1.78.0)

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
:shipit:

File: benchmark/src/main.rs [hunk 1]
```diff
@@ -9,25 +9,27 @@ use std::env;
...
 const LOG_FILE: &str = "GRPCIO_BENCHMARK_LOG_FILE";
 
+/// Benchmark QpsWorker
+///
+/// ref http://www.grpc.io/docs/guides/benchmarking.html.
+#[derive(Parser)]
+struct WorkerCli {
+    /// The port the worker should listen on. For example, 8080
+    #[arg(long)]
+    deriver_port: Option<u16>,
```
Comment (glorv) [inline benchmark/src/main.rs:27]:
```suggestion
    driver_port: Option<u16>,
```

**Review by glorv — APPROVED** [review]
LGTM

**Comment by BusyJay** [comment]
CI failure should be fixed by https://github.com/rust-lang/rust/pull/124742.

**Pull request merged**